assert/assert.go

```diff
@@ -90,6 +90,14 @@ func (assert Assert) EqFloat(actual, expected float64) {
 	}
 }
 
+// EqFloat fails the test if two floats aren't equal. NaNs are considered equal.
+func (assert Assert) EqApproximate(actual, expected, epsilon float64) {
+	delta := actual - expected
+	if !(-epsilon < delta && delta < epsilon) {
```

Comment:
If `expected` is `NaN` or infinite  then this will say that they differ (always). A special case for this at the top of the function would be useful.